docs: Add how-to to override commands in web apps (#1000)
canonical/rockcraft: docs/how-to/web-app-rocks/set-up-web-app-rock.rst

      For the ``go-framework`` extension, a deb could be needed for example to use an external command in the migration process.

Override commands
-----------------

The ``services`` stanza follows the `Pebble layer specification
<https://canonical-pebble.readthedocs-hosted.com/en/latest/reference/
layer-specification/>`_ and defines the entrypoint to your application.

> You can override the default service commands in your ``rockcraft.yaml``. 

Reviewer: ```suggestion
The ``services`` stanza follows the :external+pebble:ref:`Pebble layer specification
<layer-specification>` and defines the entrypoint to your app.
You can override the default service commands in ``rockcraft.yaml``. 
```

To override a service command, first check the default ``services``
entrypoints generated by the extension with ``rockcraft expand-extensions``

> and then adapt it to your needs. For example:

Reviewer: ```suggestion
To override a service command, check the default ``services``
entrypoints generated by the extension by running ``rockcraft expand-extensions``.
Then, adapt the command to your needs. For example:
```

.. tabs::

   .. group-tab:: Flask

      .. code-block:: yaml
         :caption: Output of ``rockcraft expand-extensions``

           ...
           services:
             flask:
               override: replace
               command: /bin/python3 -m gunicorn -c /flask/gunicorn.conf.py app:app -k [ sync ]
               startup: enabled
               after:
               - statsd-exporter
               user: _daemon_
           ...

      To limit the maximum number of pending connections in ``Gunicorn`` to 1024, add the following

>       lines to ``rockcraft.yaml``.

Reviewer: For here and the others:

```suggestion
      To limit the maximum number of pending connections in Gunicorn to 1024, add the
      following lines to your project file.
```

      .. code-block:: yaml
         :caption: rockcraft.yaml

         services:
           flask:
             command: /bin/python3 -m gunicorn -c /flask/gunicorn.conf.py app:app --backlog 1024

>                -k [ sync ]

Reviewer 2: is this `-k [ sync ]` being on a new line on purpose? If so, on my 1080 screen it might look better if you break starting on `--backlog` instead

<img width="872" height="272" alt="Image" src="https://github.com/user-attachments/assets/483b8ba2-f725-4f3d-9669-1a549df01dc8" />

   .. group-tab:: Django

      .. code-block:: yaml
         :caption: Output of ``rockcraft expand-extensions``

           ...
           services:
             django:
               override: replace
               command: /bin/python3 -m gunicorn -c /django/gunicorn.conf.py django_hello_world.wsgi:application
                 -k [ sync ]
               startup: enabled
               after:

>                - statsd-exporter

Reviewer: ```suggestion
                 - statsd-exporter
```

             user: _daemon_
         # ...

      To limit the maximum number of pending connections in ``Gunicorn`` to 1024, add the following
      lines to ``rockcraft.yaml``.

      .. code-block:: yaml
         :caption: rockcraft.yaml

         services:
           django:
             command: /bin/python3 -m gunicorn -c /django/gunicorn.conf.py django_hello_world.wsgi:application

>                --backlog 1024 -k [ sync ]

Author: yeah, I think one liner is better. Updated to it. Thanks for the comment.

   .. group-tab:: FastAPI

      .. code-block:: yaml
         :caption: Output of ``rockcraft expand-extensions``

           ...
           services:
             fastapi:
               override: replace
               command: /bin/python3 -m uvicorn app:app
               startup: enabled
               environment:
                 UVICORN_HOST: 0.0.0.0
               user: _daemon_
               working-dir: /app

      To limit the maximum number of pending connections in ``uvicorn`` to 1024, add the following
      lines to ``rockcraft.yaml``.

      .. code-block:: yaml
         :caption: rockcraft.yaml

           services:
             fastapi:
               command: /bin/python3 -m uvicorn app:app --backlog 1024

   .. group-tab:: Go

      .. code-block:: yaml
         :caption: Output of ``rockcraft expand-extensions``

           ...
           services:
             go:
               override: replace
               command: go-hello-world
               startup: enabled
               user: _daemon_
               working-dir: /app

>       To pass the argument ``--example-arg`` to the main Go application, add the following lines

Reviewer: ```suggestion
      To pass the argument ``--example-arg`` to the main Go app, add the following lines
```
